src/tiered/tiered_handle.c

```diff
@@ -113,6 +114,15 @@ __tiered_create_local(WT_SESSION_IMPL *session, WT_TIERED *tiered)
     this_tier->name = name;
     F_SET(this_tier, WT_TIERS_OP_READ | WT_TIERS_OP_WRITE);
 
+    /*
+     * We should have a better way to handle the initial case of setting the object number.
+     */
+    if (((WT_BTREE *)tiered->iface.handle)->bm != NULL) {
+        WT_WITH_DHANDLE(
+          session, &tiered->iface, ret = __wt_btree_switch_object(session, tiered->current_id, 0));
+        WT_ERR(ret);
```

Comment:
Instead of reaching into the btree to figure out if the block manager has been created, could we let `__wt_btree_switch_object()` handle this?  

Alternately we could add a flag on `WT_TIERED` to indicate that we are initializing. Or perhaps the inverse, a flag that says our BTREE is setup.

Reply:
I rewrote as you suggested.